app/controllers/publishers/omniauth_callbacks_controller.rb

```diff
@@ -182,17 +181,46 @@ def register_vimeo_channel
       redirect_to home_publishers_path, notice: t("shared.channel_created")
     end
 
+    def register_reddit_channel
+      reddit_auth_hash = request.env['omniauth.auth']
+      @channel = current_publisher.channels.new(verified: true)
+      @channel.details = RedditChannelDetails.new(
+        name: reddit_auth_hash.info.name,
+        reddit_channel_id: reddit_auth_hash.uid,
+        auth_provider: reddit_auth_hash.provider,
+        thumbnail_url: reddit_auth_hash.extra.raw_info.icon_img,
+        channel_url: "https://www.reddit.com/user/#{reddit_auth_hash.info.name}",
+        nickname: reddit_auth_hash.info.name,
+      )
+
+      existing_channel = Channel.joins(:reddit_channel_details).where("reddit_channel_details.reddit_channel_id": reddit_auth_hash.uid).first
```

Comment:
Let's ensure that the channel is verified when were finding the existing channel